src/itemlist_formaction.cpp

```diff
@@ -361,6 +361,21 @@ void itemlist_formaction::process_operation(operation op, bool automatic, std::v
 			v->show_error(strprintf::fmt(_("Error: couldn't mark feed read: %s"), e.what()));
 		}
 		break;
+	case OP_MARKFEEDREAD_UNTIL_ARTICLE: 
+		LOG(level::INFO, "itemlist_formaction: marking feed read until current article");
+		v->set_status(_("Marking feed read until current article..."));   
+		if (itemposname.length() > 0 && itempos < visible_items.size()) {
+			for (unsigned int i=0; i<=itempos&&i<visible_items.size(); ++i) {
+				if (visible_items[i].first->unread()) {
+					visible_items[i].first->set_unread(false);
+					v->get_ctrl()->mark_article_read(visible_items[i].first->guid(), true);
+				}
+			}  
+			if (!v->get_cfg()->get_configvalue_as_bool("show-read-articles")) f->set("itempos","0");
```

Comment:
In new code, we prefer to always use curly braces, so please change that line to:

```c++
if (!v->get_cfg()->get_configvalue_as_bool("show-read-articles")) {
        f->set("itempos","0");
}
```

Older code hasn't been updated to follow that convention—sorry about that.

Reply:
OK. No problem.